### DISCUSSION TITLE
Which distros to support

### DISCUSSION THREAD

**Discussion by maliberty** [The-OpenROAD-Project/OpenROAD#3350] [General]
Currently OR CI is built with
* Local centos7 gcc8
* Local centos7 gcc8 without GUI
* Docker centos7 gcc8
* Docker ubuntu20.04 gcc9

PRs additionally have GH actions that test:
* macOS (x86)
* c++20 (using clang-16)

Centos7 is getting quite dated and I think it is time to replace it with a more modern distro.  #3344 shows an example problem where qt5charts is not available in this older distro.  I think Ubuntu 22 or Debian 11 are good contenders.

I'd like community input as to what distro others are using with OR to help guide our choice.  @antonblanchard @gadfort @QuantamHD @rovinski @msaligane @oharboe @donn @nayiri-k

The default compiler version would be updated to match the distro default.

FYI @vvbandeira @tspyrou

**Comment by QuantamHD** [comment]
Google is on Debian test called [Glinux](https://en.wikipedia.org/wiki/GLinux). Ubuntu 22 or Debian 11 seem like reasonable choices to me.

But we use bazel to build everything from source so the biggest issue for me is if we start requiring very new versions of widely used libraries like Boost, because then I have to upgrade all of Google to use that version of software which can be a big pain.

Google has a one version policy for any piece of software in the universe, and it's all stored in a monorepo.

**Reply by maliberty** [reply]
I would expect not to require anything more recent than what the distro itself provides without strong motivation.

**Reply by QuantamHD** [reply]
I think other consideration I would weigh is trying to get the best C++20 support we can out of the compiler version in the new distro.

**Comment by donn** [comment]
I use Nix as a build system, so as long as the CMakeFiles are generic enough to compile on both macOS, Ubuntu and CentOS I have no issues with the specific versions of the operating system.

**Reply by maliberty** [reply]
Supporting old versions is work as I noted below to Austin.  Do you care about centos7 or 8?

**Reply by donn** [reply]
No- shorter term, when it's time to migrate OpenLane 1's operating system, I believe we're moving to Rocky Linux 9 (successor to CentOS), so we do need that, but nothing older. I would prefer a timeframe for this though as moving OpenLane to RL9 is not trivial. 

Longer term (OpenLane 2,) Nix is flexible enough that the one problem I would have would be Debian-only assumptions made in the CMakeFiles.

**Comment by rovinski** [comment]
CentOS 7 is EOL on 2024-06-30. I know of machines that are still running it, so maybe it would be a good idea to hold off removing support until then.

**Reply by maliberty** [reply]
There is a cost to doing so.  I don't think we have to wait until the last possible moment.  We would have to have (1) no gui (2) gui but without qcharts and (3) full gui.  I'm not clear it is worth the extra effort for that.   Do you have a lot of centos7 machines?

**Reply by rovinski** [reply]
Is it possible there is another repo available that has qtcharts? I'm pretty sure that's what was already being done with rhel7.

**Reply by maliberty** [reply]
No idea

**Reply by rovinski** [reply]
I just checked, and my primary server at Cornell runs CentOS 7, so it's important to me until at least the end of the summer. I will mention to them they should upgrade, but no guarantees on when that happens.

**Reply by oharboe** [reply]
We use the Docker image with GUI and Ubuntu 22.04 or newer. I would be happy with a policy of supporting 22.04+docker w GUI and welcoming pull requests that fix, but do not complicate or break supported distributions.

I would like ORFS to include a batch file example that launches OpenROAD and the GUI with Docker. I intend to submit a.pull request. The docker GUI is still crashing, which is why we use Ubuntu 22.04 + local install is also used for the debugging and.profiling use case. On build servers we use only docker.

**Reply by gadfort** [reply]
We have some centos7 / rhel7 machines I'd like to continue to support for a while. I would be okay with deprecating GUI support on that platform and when later this year deprecate support for the centos7 platform all together. I'm not sure what the minimum platform should be, but it would be great if we would be able to support RHEL8 /centos8 without much effort.

**Comment by olofk** [comment]
As a data point, I wanted to mention that I just got it running on Gentoo, depending only on system packages except for or-tools
